vehicle/security/vehicle.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
+        <record id="group_vehicle_user" model="res.groups">
+            <field name="name">User</field>
+            <field name="category_id" ref="module_vehicle_category"/>
+            <field name="implied_ids" eval="[(4, ref('base.group_user'))]"/>
+        </record>
+        <record id="group_vehicle_manager" model="res.groups">
+            <field name="name">Administrator</field>
+            <field name="category_id" ref="module_vehicle_category"/>
+            <field 
+                name="implied_ids"
+                eval="[(4, ref('vehicle.group_vehicle_user'))]"
+            />
+            <field
+                name="users"
+                eval="[(4, ref('base.user_root')), (4, ref('base.user_admin'))]"
+            />
+        </record>
+
+        <record id="vehicle_edit_form_rule" model="ir.rule">
+            <field name="name">Vehicles User Rule</field>
+            <field name="model_id" ref="model_vehicle_vehicle" />
+            <field name="global" eval="True" />
+            <field name="domain_force">[('user_id','=',user.id)]</field>
+            <field name="perm_read" eval="0" />
+            <field name="perm_create" eval="0" />
+            <field name="perm_write" eval="1" />
+            <field name="perm_unlink" eval="0" />
+            <field name="groups" eval="[(4, ref('vehicle.group_vehicle_user'))]" />
+        </record>
+    </data>
```

Comment:
Record rule that invalidates the restricted one is missing (the typical `[(1, '=', 1)]`]). If you test with more than one user, you'll realize that nobody can change the others vehicles